File: Array/128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)
        maxLen = 0
        for num in nums_set:
            if num-1 not in nums_set:
                count = 1
                while num+count in nums_set:
                    count+=1
                
                maxLen = max(maxLen, count)

        return maxLen

File: JavaScript/128-longest-consecutive-sequence/longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    let numsSet = new Set(nums);
    let maxConsec = 0;
    for(let num of numsSet) {
       
        if(!numsSet.has(num-1)){
            let count = 1;
            let currNum = num;
            while(numsSet.has(currNum+1)) {
                count++;
                currNum++;
            }
            maxConsec = Math.max(maxConsec, count);
        }
    }
    return maxConsec;

};